tests/monitoring/monitoring.go

```diff
@@ -638,29 +637,85 @@ var _ = Describe("[Serial][sig-monitoring]Prometheus Alerts", Serial, decorators
+					{
+						ResourceName: virtHandler.deploymentName,
+						ResourceType: "DaemonSet",
+						Patch:        `{"spec":{"template":{"spec":{"nodeSelector":{"kubernetes.io/hostname":"does-not-exist"}}}}}`,
+						Type:         v1.StrategicMergePatchType,
+					},
+				},
+			}
+
+			kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(kv)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() string {
+				vh, err := virtClient.AppsV1().DaemonSets(originalKv.Namespace).Get(context.Background(), virtHandler.deploymentName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return vh.Spec.Template.Spec.NodeSelector["kubernetes.io/hostname"]
+			}, 90*time.Second, 5*time.Second).Should(Equal("does-not-exist"))
+
+			return kv
+		}
+
+		setEmulation := func(status bool) *v1.KubeVirt {
+			originalKv := util.GetCurrentKv(virtClient)
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			kv.Spec.Configuration.DeveloperConfiguration.UseEmulation = status
+
+			kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(kv)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() bool {
+				kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return kv.Spec.Configuration.DeveloperConfiguration.UseEmulation
+			}, 90*time.Second, 5*time.Second).Should(Equal(status))
+
+			return kv
+		}
+
+		restoreVirtHandler := func(kv *v1.KubeVirt) {
+			originalKv := util.GetCurrentKv(virtClient)
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			kv.Spec.CustomizeComponents = v1.CustomizeComponents{}
+			kv, err = virtClient.KubeVirt(kv.Namespace).Update(kv)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
 		It("KubeVirtNoAvailableNodesToRunVMs should be triggered when there are no available nodes in the cluster to run VMs.", func() {
+			By("Scaling down virt-handler")
+			kv := disableVirtHandler()
+
 			By("Removing the kubevirt.io/schedulable label from all nodes")
 			nodes, err := virtClient.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
 			Expect(err).ToNot(HaveOccurred())
-
 			removeSchedulableLabelFromNodes(virtClient, nodes)
 
-			By("Verifying KubeVirtNoAvailableNodesToRunVMs alert exists")
-			verifyAlertExist("KubeVirtNoAvailableNodesToRunVMs")
+			By("Verifying KubeVirtNoAvailableNodesToRunVMs alert exists if emulation is disabled")
+			setEmulation(false)
+			verifyAlertExistWithCustomTime("KubeVirtNoAvailableNodesToRunVMs", 10*time.Minute)
+
+			By("Verifying KubeVirtNoAvailableNodesToRunVMs alert does not exist if emulation is enabled")
+			setEmulation(true)
+			waitUntilAlertDoesNotExist("KubeVirtNoAvailableNodesToRunVMs")
+
+			By("Restoring virt-handler and emulation")
+			restoreVirtHandler(kv)
+			setEmulation(false)
```

Comment:
It may well be the emulation was `true` previously. This is a redundant line.

Reply:
im now saving the original emulation status and restoring it